maas-api.yaml

```diff
@@ -1,555 +1,677 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InputError'
+
+  /routing/grouped:
+    post:
+      tags:
+        - Trip Planning
+      summary: A-to-B trips (grouped)
+      description:
+        Calculates door-to-door trips for the specified mode(s) which are grouped by their similarity (e.g., same public transport trip at a later time)
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RoutingInput'
+
+      responses:
+        200:
+          description: Successful response. Can include many trips.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GroupedRoutingResponse'
+        400:
+          description: Bad request, most likely due to issues with the input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InputError'                
 
-  /waypoint.json:
+  /waypoint:
     post:
       tags:
         - Trip Planning
-      summary: A-to-B-to-C trip
+      summary: A-to-B-via-C trip
       description: |
-        Calculates a single A-to-B-to-C trip where transport modes can be customised per segment.
+        Calculates a single A-to-B-via-C trip where transport modes can be customised per segment.
 
         Of the provided trip pattern items in the `waypoints` input, exactly one should have the
         `time` field set.
-
-        Note: If you require alternatives for travelling between one waypoint and the next, use the `/ttp/` endpoint instead.
-      parameters:
-        - name: input
-          in: body
-          required: true
-          schema:
-            type: object
-            properties:
-              waypoints:
-                type: array
-                items:
-                  $ref: '#/definitions/TripInputPattern'
-              config:
-                $ref: '#/definitions/ConfigInput'
-            required:
-              - config
-              - waypoints
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                waypoints:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/TripInputPattern'
+                config:
+                  $ref: '#/components/schemas/ConfigInput'
+              required:
+                - config
+                - waypoints
       responses:
         200:
-          description: Successful response. Only one trip returned.
-          schema:
-            $ref: '#/definitions/RoutingResponse'
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Trip'
 
   /trip/{id}:
```

Comment:
other endpoints like `/locations/{id}` and `/regions/{id}` are plural.  should we use `/trips/{id}` here?

Reply:
Good catch, will change that.